### ISSUE TITLE
click>=8.0 causes flow to be broken

### ISSUE THREAD

**Issue by zexuan-zhou** [Netflix/metaflow#537]
Sorry don't have a simple testing script to reproduce the error but I'm seeing this for my flow
```
    Internal error
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/metaflow/cli.py", line 930, in main
    start(auto_envvar_prefix='METAFLOW', obj=state)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1134, in __call__
    return self.main(args, kwargs)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1663, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 927, in make_context
    self.parse_args(ctx, args)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1376, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 2352, in handle_parse_result
    value = self.process_value(ctx, value)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 2308, in process_value
    value = self.type_cast_value(ctx, value)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 2295, in type_cast_value
    return convert(value)
  File "/opt/conda/lib/python3.6/site-packages/click/types.py", line 75, in __call__
    return self.convert(value, param, ctx)
  File "/opt/conda/lib/python3.6/site-packages/metaflow/includefile.py", line 230, in convert
    value = os.path.expanduser(value)
  File "/opt/conda/lib/python3.6/posixpath.py", line 235, in expanduser
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not function
```
Once I downgrade click to 7.0.0 it works

**Comment by savingoyal** [comment]
@zexuan-zhou Which version of Metaflow are you using? We addressed this in [`2.2.12`](https://docs.metaflow.org/introduction/release-notes#handle-regression-with-click-greater-than-8-0-x)

**Comment by zexuan-zhou** [comment]
> @zexuan-zhou Which version of Metaflow are you using? We addressed this in [`2.2.12`](https://docs.metaflow.org/introduction/release-notes#handle-regression-with-click-greater-than-8-0-x)

`2.2.8`. Thank you!
